frontend/packages/dev-console/src/components/import/advanced/ProgressiveList.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import FormSectionDivider from '../section/FormSectionDivider';
+import './ProgressiveList.scss';
+
+export const getGeneratedText = (text, items: string[], callback) => {
+  if (!items || items.length === 0) {
+    return null;
+  }
+  return (
+    <p>
+      {text}
+      {items.map((opt, index) => {
+        let preText = ' ';
+        let postText = '';
+        if (items.length - 1 === index) {
+          preText = items.length !== 1 ? ' and ' : ' ';
+          postText = '.';
+        } else {
+          postText = items.length - 2 !== index ? ',' : '';
+        }
+        return (
+          <React.Fragment key={opt}>
+            {preText}
+            <button
+              className="btn btn-link odc-progressive-list__button"
+              type="button"
+              onClick={(event: React.MouseEvent<HTMLButtonElement>) => {
+                const target = event.target as HTMLButtonElement;
+                callback(target.innerText);
+              }}
+            >
+              {opt}
+            </button>
+            {postText}
+          </React.Fragment>
+        );
+      })}
+    </p>
+  );
+};
+
+type ProgressiveListProps = {
+  visibleItems: string[];
+  onVisibleItemsChanged: (item: string) => void;
+  text: string;
+};
+
+export const ProgressiveList: React.FC<ProgressiveListProps> = ({
+  visibleItems,
+  children,
+  onVisibleItemsChanged,
+  text,
+}) => {
+  const items = [];
+  return (
+    <React.Fragment>
+      {React.Children.map(children, (child: React.ReactElement, index: number) => {
+        const { name } = child.props;
+        const isNameInVisibleItems = visibleItems.includes(name);
+        if (!isNameInVisibleItems) {
+          items.push(name);
+        }
+        return isNameInVisibleItems ? (
+          <React.Fragment>
+            {child}
+            {index !== React.Children.count(children) - 1 && <FormSectionDivider />}
+          </React.Fragment>
+        ) : null;
+      })}
+      {getGeneratedText(text, items, onVisibleItemsChanged)}
+    </React.Fragment>
+  );
+};
+
+type AdvancedSectionProps = {
+  name: string;
+};
+export const ProgressiveListItem: React.FC<AdvancedSectionProps> = ({ children }) => {
```

Comment:
Move this out to its own file. We are following one component per file with default export.